### PR TITLE
OKTA-397356: add unit test for recover password

### PR DESCRIPTION
**Pull request by bryanapellanes-okta**
- Unit tests for recover password method

### CODE REVIEW

File: src/Okta.Idx.Sdk/Extensions/IdxClientExtensions.cs [hunk 1]
```diff
@@ -1,26 +1,26 @@
-﻿// <copyright file="IdxClientExtensions.cs" company="Okta, Inc">
```
Comment (bryanapellanes-okta) [inline src/Okta.Idx.Sdk/Extensions/IdxClientExtensions.cs:1]: My IDE decided to "normalize" line endings here.  This file did not actually get modified in any functional way.

**Review by laura-rodriguez — COMMENTED** [review]
We need to add more tests to cover all the cucumber scenarios. I suggest adding the following UTs:

Scenario 3.1.1
OktaClientShould.SelectsEmailToRecoverPassword
 - Check Status is AwaitingAuthenticatorVerification
OktaClientShould.VerifyEmailToRecoverPassword
  -  Check Status is AwaitingPasswordReset
OktaClientShould.ResetPassword
  - Check Status is Success and Tokens !=null
 
Scenario 3.1.2
OktaClientShould.FailWithInvalidEmail
- Check an exception with the expected message is thrown.

File: mocks/03-self-service-password-recovery/reset-password-challenge-answer-response.json [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+            },
+            {
+                "name": "interaction_code",
+                "required": true,
+                "value": "T6J_vuM3rcvXxU8Awrx2uxE56bBXMDvcBZ9Z03gz-zk"
+            },
+            {
+                "name": "client_id",
+                "required": true,
+                "value": "0oatzfskmLm4faAaQ5d6"
```
Comment (laura-rodriguez) [inline mocks/03-self-service-password-recovery/reset-password-challenge-answer-response.json:60]: Let's mask this value as well.

File: mocks/03-self-service-password-recovery/verify-email-challenge-answer-response.json [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+            },
+            {
+                "name": "interaction_code",
+                "required": true,
+                "value": "T6J_vuM3rcvXxU8Awrx2uxE56bBXMDvcBZ9Z03gz-zk"
+            },
+            {
+                "name": "client_id",
+                "required": true,
+                "value": "0oatzfskmLm4faAaQ5d6"
```
Comment (laura-rodriguez) [inline mocks/03-self-service-password-recovery/verify-email-challenge-answer-response.json:60]: Let's mask this value as well.

File: src/Okta.Idx.Sdk.UnitTests/IdxClientShould.cs [hunk 4]
```diff
@@ -2372,7 +2372,2234 @@ public async Task CallInteractAndIntrospectOnGetIdentityProviders()
...
+            },
+            {
+                ""name"": ""interaction_code"",
+                ""required"": true,
+                ""value"": ""T6J_vuM3rcvXxU8Awrx2uxE56bBXMDvcBZ9Z03gz-zk""
+            },
+            {
+                ""name"": ""client_id"",
+                ""required"": true,
+                ""value"": ""0oatzfskmLm4faAaQ5d6""
```
Comment (laura-rodriguez) [inline src/Okta.Idx.Sdk.UnitTests/IdxClientShould.cs:4246]: Let's mask this value.

**Review by laura-rodriguez — APPROVED** [review]
LGTM

**Pull request merged**